Clinical-Genomics/cg
# Add analysis link balsamic fastqs

This PR adds the new feature of linking and concatenating balsamic fastq files.

How to test:
0. install on stage. 
Testdata:
- Family: loyalkiwi
- Samples: ACC5152A1, ACC5152A2, ACC5152A3
- For those samples set data_analysis type to 'Balsamic', 'MIP', 'MIP + Balsamic' respectively
- Extra family: testyanimal, if not in stage db create: choose id, internal_id = testyanimal, name = 3-UA05fam2, priority = 1, panels = OMIM-AUTO, customer_id = 58
- cg add relationship -s affected loyalkiwi ACC5152A1
- cg add relationship -s affected loyalkiwi ACC5152A2
- cg add relationship -s affected loyalkiwi ACC5152A3
- cg add relationship -s affected testyanimal ACC5152A1

Note: pay attention to the expected log messages during the executing of the test cases:
- ```<sample_id>: link FASTQ files```
- ```linking: /mnt/hds/proj/bioinfo/DEMUX/<file_name> -> /mnt/hds/proj/bioinfo/STAGE/BALSAMIC_analysis/<case_id/fastq/<file_name>``` for all fastq files for that sample
- ```Concatenation in progress for sample <sample_id>.```
- ```Concatenating: <fastq files> -> concatenated_<sample_id_>_XXXXXX_R_1.fastq.gz``` all fastq files for read 1
- ```Concatenating: <fastq files> -> concatenated_<sample_id_>_XXXXXX_R_2.fastq.gz``` all fastq files for read 2
- ```Concatenation file size check successful!```
- repeat for all samples in test case

1. Test Case 1: link and concatenate all samples in a family: ```cg analysis link -f loyalkiwi```
Expected outcome: 
- all three samples are linked in ```/mnt/hds/proj/bioinfo/STAGE/families/<case-id>/wes/<sample_id>/fastq/```
- samples ```ACC5152A1``` and ```ACC5152A3``` are also linked in ```/mnt/hds/proj/bioinfo/STAGE/BALSAMIC_analysis/<case-id>/fastq/``` Check logs/messages because these files are removed at the end!
- Concatenated files for samples ```ACC5152A1``` and ```ACC5152A3``` can be found in ```/mnt/hds/proj/bioinfo/STAGE/BALSAMIC_analysis/<case-id>/fastq/```
- linked files for ```ACC5152A1``` and ```ACC5152A3``` are removed from ```/mnt/hds/proj/bioinfo/STAGE/BALSAMIC_analysis/<case-id>/fastq/```

2. Test Case 2: link and concatenate one sample in a family: ```cg analysis link ACC5152A3```
Expected outcome: 
- sample is linked in ```/mnt/hds/proj/bioinfo/STAGE/families/loyalkiwi/wes/ACC5152A3/fastq/```
- sample is also linked in ```/mnt/hds/proj/bioinfo/STAGE/BALSAMIC_analysis/loyalkiwi/fastq/``` Check logs/messages because these files are removed at the end!
- Concatenated files for sample ```ACC5152A3``` can be found in ```/mnt/hds/proj/bioinfo/STAGE/BALSAMIC_analysis/loyalkiwi/fastq/```
- linked files for ```ACC5152A3``` are removed from ```/mnt/hds/proj/bioinfo/STAGE/BALSAMIC_analysis/loyalkiwi/fastq/```

3. Test Case 3: link and concatenate one sample that's in two families: ```cg analysis link ACC5152A1```
Expected outcome: 
- sample is linked in ```/mnt/hds/proj/bioinfo/STAGE/families/loyalkiwi/wes/ACC5152A1/fastq/```
 sample is linked in ```/mnt/hds/proj/bioinfo/STAGE/families/testyanimal/wes/ACC5152A1/fastq/```
- sample is linked in ```/mnt/hds/proj/bioinfo/STAGE/BALSAMIC_analysis/loyalkiwi/fastq/``` Check logs/messages because these files are removed at the end!
- sample is linked in ```/mnt/hds/proj/bioinfo/STAGE/BALSAMIC_analysis/testyanimal/fastq/``` Check logs/messages because these files are removed at the end!
- Concatenated files for sample ```ACC5152A1``` can be found in ```/mnt/hds/proj/bioinfo/STAGE/BALSAMIC_analysis/loyalkiwi/fastq/```
- Concatenated files for sample ```ACC5152A1``` can be found in ```/mnt/hds/proj/bioinfo/STAGE/BALSAMIC_analysis/testyanimal/fastq/```
- linked files for ```ACC5152A1``` are removed from ```/mnt/hds/proj/bioinfo/STAGE/BALSAMIC_analysis/loyalkiwi/fastq/```
- linked files for ```ACC5152A1``` are removed from ```/mnt/hds/proj/bioinfo/STAGE/BALSAMIC_analysis/testyanimal/fastq/```

4 Test case 4: cli command to remove balsamic fastq files
- run ``` cg analysis remove_fastq -f loyalkiwi```
Expected outcome folder ```/mnt/hds/proj/bioinfo/STAGE/BALSAMIC_analysis/loyalkiwi/fastq/``` is removed.
